mpqp/core/instruction/gates/native_gates.py

```diff
@@ -868,6 +946,40 @@ def qiskit_gate(cls):
 
         return UGate
 
+    @classproperty
+    def cirq_gate(cls):
+        from cirq.circuits.qasm_output import QasmUGate
+        from cirq.ops.common_gates import ry as cirq_ry, rz as cirq_rz
+        from cirq.ops.global_phase_op import GlobalPhaseGate
+        from cirq.ops.raw_types import Qid
+
+        class CirqUGate(QasmUGate):  # pyright: ignore[reportUntypedBaseClass]
```

Comment:
The class will be redifined each time we call the property. Isn't it better if we define the class outside and only return it when needed ?